### DISCUSSION TITLE
multiple targets with same names

### DISCUSSION THREAD

**Discussion by huand** [cpm-cmake/CPM.cmake#234] [Q&A] [answered]
Hello, when adding several libs with CPMAddPackage, I get the following error because several libraries defines the "uninstall" target.
Example of error:

```
CMake Error at /home/hubert/.cache/CPM/eigen/36d46e474d95627b67830e4f72c75dd26d7e34f6/CMakeLists.txt:644 (add_custom_target):
  add_custom_target cannot create target "uninstall" because another target
  with the same name already exists.  The existing target is a custom target
  created in source directory
  "/home/hubert/Documents/XXXXXX
```

any idea how to get around this?

**Comment by JohelEGP** [comment]
CMake targets are required to be uniquely named. The solution is for the mantainers of CMake buildsystems to add a unique prefix to their targets. If eigen isn't doing that, then maybe it's a new addition that should be reported and fixed, or their CMake buildsystem isn't made to be used through `find_package` (perhaps it exists solely to be used by CI and not for users to easily depend on their library).

**Comment by TheLartians** [comment, marked as answer]
You have to define your own targets if a library has bad CMake support. See the [eigen snippet](https://github.com/cpm-cmake/CPM.cmake/wiki/More-Snippets#eigen) for an example how to do that.
